x-pack/plugins/files/common/types.ts

```diff
@@ -151,9 +168,10 @@ export type FileMetadata<Meta = unknown> = Required<
     FileKind: string;
 
     /**
-     * User-defined metadata
+     * User-defined metadata and some pre-defined system fields that can be used
+     * for searching and filtering.
      */
-    Meta?: Meta;
+    Meta?: Meta & FileSystemMetadata;
```

Comment:
This was added to the `Meta` field because I was running into the 1000 fields upper limit for the SO index

[Long link](https://s3.amazonaws.com/buildkiteartifacts.com/e0f3970e-3a75-4621-919f-e6c773e2bb12/0fda5127-f57f-42fb-8e5a-146b3d535916/018414ac-aae3-4538-ba57-da03925732e1/018414ae-6532-48e0-a83b-cb5a2f0003ed/target/test_failures/018414ae-6532-48e0-a83b-cb5a2f0003ed_66d2dcc77dabdd89d1561198a4fad624.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LR7YSL7KF%2F20221031%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221031T111709Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEIj%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJGMEQCIAFqLkzjpkMwseRAXLVLfjjsYKKsqlL5P22CRJrbAvN1AiA1AcYRcjv5S%2BJc%2F03eN7Gf4zK6VQxmiD%2FbV9TAQxCEAirMBAhxEAAaDDAzMjM3OTcwNTMwMyIM9nvSXc48xZrLkYTjKqkEoPC589qtc5%2FnoGbBtMU4TmSybXF0kUnj2R1y7Te2Vl5JhyVa1PUP35c7xP3ilpR9rBlVadKPH6myUuS7URBBJNr8YakdeXWObQV9JdSik7VyaLked7gEzIsg87pdoZOvMHSFUsQkizSHfBsdldHlMnKEm%2F3Ws%2B6zAeHhhpoJuOONXIKmDv42a2HZmxLS9pTUGSKQuDQUfBMDpW8V7JL9WNcR%2FrYwucoNosEbgyKr8Gz43%2Bc8M9sUuZ2C5IG1sXEfqg%2B8WKuWi7o2GqfS3YT503zX3Xlb8v0czau7quiy1EctzyKD93jUQmt6Jc1H%2BlKyHp%2FTVjy0iwFx9rU%2BwOzrvC%2FqEpQyVYcczpDDonFGcAjTBGVZLHQA4dY1JTxAmzlKo6va8XyKaSJDNQZeYM6pwAuoHhoA7ECrDi%2BoQl8dLc8rcDeKcWJQe1bkiP3x7OUqQZAFc%2FP4nEwo42qeAlhTqqADnyLYeQElFjwn41qPXTB99DJNqYu6qsp8PiaQHabV%2BypCDj20Z4EgW0QY6YJcGuMDfGZlZa%2FNbtPBQZKYYOUihktQygWSlb8FxbFrTUKSl5N%2FDLKC1l%2BpM6XP3jO%2FTmB%2BxIO5zD8ajLiPSq60YjAeq2kDU2Ay9XVqMgNLZEziuZ4sbdGWp%2BNWaP3EVBElHDVRHYIUW6T4cnZ5MEfhIxzVo4DlV8%2FgCxTPPKXDcgwuDKNDagvze7Y%2FSwPILkmNxy00rknTfbDbhTCk%2BP2aBjqqAYH7mxnl35N6tPHSj0kmCrgfFtvDHJNrPgMJHug%2FevV3JkdWopG2DeO%2B2MUHqD2nKIb8JEu82aE3FPqJkcpnhbNc49ITktnnZgSMO6irylHmfKWndBjtAmoV28ManzxZOLEocI7l38wcibpEqlx6TN22hf6HBbGfeXIE1KrlXbO7L4S74nDGjRr%2FlOtjUnWBHCiHAlwh3%2B2T6SsGLEp%2BktIiEOSJFIYwNRss&X-Amz-SignedHeaders=host&X-Amz-Signature=837b531ce8a7615f9ed0520314c54ebd2c828267b465f7ffc00985ac7991a2ce)

Happy to use the regular declaration for new fields, but since we already have a flattened field that kind of fits this use case, I pivoted the implementation to use that instead. cc @vadimkibana

Reply:
Also notified core about this limit being hit in our functional tests

Reply:
A there any downsides of storing it under meta? Are these still searchable/aggregatable / filterable?